results-processor/Dockerfile

```diff
@@ -20,4 +20,7 @@ ADD requirements.txt /app/
 RUN pip install -r requirements.txt
 
 ADD . /app/
-CMD exec gunicorn -b :$PORT -t 3600 main:app
+# The number of workers should always be 2: one for processing tasks, the other
```

Comment:
Doesn't this make health checks pretty useless? The worker responsible for servicing "real" requests could be unhealthy and we would never find out.

Reply:
We kinda have to have two workers (to clarify, they are simply two processes running the same code) because of the way WSGI works (WSGI is very similar to the traditional CGI and has a synchronous design). I could rewrite the whole thing from ground up using `asyncio` to make it asynchronous, but that'd make it substantially more complicated.

The health checks are indeed not ideal. One of the workers can die or hang while processing the results without releasing the lock and the other worker will forever return busy, leaving the instance stuck at this state.

We don't need to rewrite the whole thing asynchronous to solve this problem. As suggested by @lukebjerring , in liveness checks, we can respond an error if the lock hasn't been modified for a while, and AppEngine will restart the instance. As for the threshold, we can use 1 hour as a starter (which should be enough to process one full report); later we can make some time-consuming external calls (namely `gsutil rsync`) non-blocking and touch the file every few minutes while waiting for the calls to complete, so that we can use a smaller threshold. (I'd like to leave the latter improvement in a future PR.)